packages/dma-library/src/operations/ajna/adjust-risk-down.ts

```diff
@@ -0,0 +1,105 @@
+import { ajnaAdjustDownOperationDefinition } from '@deploy-configurations/operation-definitions/ajna/adjust-down'
+import { MAX_UINT } from '@dma-common/constants'
+import { actions } from '@dma-library/actions'
+import {
+  IOperation,
+  WithAjnaBucketPrice,
+  WithAjnaStrategyAddresses,
+  WithCollateralAndWithdrawal,
+  WithDebt,
+  WithFlashloan,
+  WithProxy,
+  WithSwap,
+} from '@dma-library/types'
+import { FlashloanProvider } from '@dma-library/types/common'
+import BigNumber from 'bignumber.js'
+
+type AjnaAdjustRiskDownArgs = WithCollateralAndWithdrawal &
+  WithDebt &
+  WithSwap &
+  WithFlashloan &
+  WithProxy &
+  WithAjnaStrategyAddresses &
+  WithAjnaBucketPrice
+
+export type AjnaAdjustRiskDownOperation = ({
+  collateral,
+  debt,
+  swap,
+  flashloan,
+  proxy,
+  addresses,
+  price,
+}: AjnaAdjustRiskDownArgs) => Promise<IOperation>
+
+export const adjustRiskDown: AjnaAdjustRiskDownOperation = async ({
+  collateral,
+  debt,
+  swap,
+  flashloan,
+  proxy,
+  addresses,
+  price,
+}) => {
+  const setDebtTokenApprovalOnPool = actions.common.setApproval({
+    asset: debt.address,
+    delegate: addresses.pool,
+    amount: flashloan.amount,
+    sumAmounts: false,
+  })
```

Comment:
approval need to be set up after swap. The amount is unknown before swap.

Reply:
fixed